### ISSUE TITLE
Don't trigger ComposeUnstableCollections when unstable collection is used in preview

### ISSUE THREAD

**Issue by carstenhag** [slackhq/compose-lints#351]
I have something like this in use:

```kotlin
private class ButtonParameter : PreviewParameterProvider<List<Any?>> {
    override val values: Sequence<List<Any?>>
        get() = sequenceOf(
            listOf<Any?>("Very, very long button text that takes two lines", R.drawable.ic_charge, true, false),
            listOf<Any?>("Button", null, true, false),
...

fun ButtonSecondaryPreview(@PreviewParameter(ButtonParameter::class) properties: List<Any?>)
```

The linter complains about this and in normal code this would be great, but I don't really see the point of changing it in previews. Just takes a bit of time, but it's not time spent effectively apart from making the linter happy.

**Comment by ZacSweers** [comment]
PR is welcome 👍

**Comment by ZacSweers** [comment]
thinking more on this, this is trivial enough that if you care about the lint to be enabled, you can also just use `ImmutableList`/`persistentListOf` for these cases
